bottom/protocol.py

```diff
@@ -1,29 +1,38 @@
 import asyncio
 import logging
 from bottom.unpack import unpack_command
+from typing import Optional, TYPE_CHECKING
+
+if TYPE_CHECKING:
+    from bottom.client import Client  # noqa
+
 # Always write the full \r\n per spec, but accept \n when reading
 DELIM = b"\r\n"
 DELIM_COMPAT = b"\n"
 log = logging.getLogger('bottom')
 
 
 class Protocol(asyncio.Protocol):
-    def __init__(self, client=None):
-        self.client = client
-        self.closed = False
-        self.transport = None
-        self.buffer = b""
+    client = None  # type: Client
+    transport = None  # type: asyncio.WriteTransport
+
+    def __init__(self, client: 'Optional[Client]' = None) -> None:
+        if client is not None:
+            self.client = client
+        self.closed = False  # type: bool
```

Comment:
Are you sure? If they weren't defined in `__init__`, all instances would share the same objects, which could cause bugs.